src/triage/component/catwalk/utils.py

```diff
@@ -62,6 +63,30 @@ def retry_if_db_error(exception):
 db_retry = retry(**DEFAULT_RETRY_KWARGS)
 
 
+def load_query_if_needed(config_component):
+    """ Load the cohort or label query from a file
+
+    Args:
+        config_component (dict) A cohort or label config
+
+    Returns: None
+    """
+    if "filepath" in config_component:
+        if "query" in config_component:
+            raise ValueError(
+                "Both filepath and query are present. Please provide only one."
+            )
```

Comment:
Should this check exist both here and in the experiment validation, or just in the validation? If we think it should be in both places, should we use the same logic for the check in both (in the validator it looks at the length of the set of keys)?

Reply:
The validator calls this function, so it should raise the error, but it won't have the typical validator boilerplate. I would lean towards changing that being part of implementing a new config loader class. Thoughts?

Reply:
Ah, looking over my changes to the validator, I see the set thing... Yeah, that's hard to read, but it's a bit more strict: It checks that exactly 1 of the two keys exists, which will prevent other kinds of failures (i.e., both keys are missing).

Reply:
I might actually lean towards something even more strict there: Rather than using set, ensure that there is exactly 1 key matching either of those names, which would prevent something that has 2 filenames or 2 queries from passing.

Reply:
IOW, I think the validator and the runtime logic do not need to be the same

Reply:
I've removed the runtime check in favor of a warning level log